src/lib/mongoBackend/safeBsonGet.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+*
+* Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Fermin Galan Marquez
+*/
+
+#include "mongoBackend/safeBsonGet.h"
+#include "logMsg/logMsg.h"
+
+using namespace mongo;
+
+/* ****************************************************************************
+*
+* getObjectField -
+*
+*/
+BSONObj getObjectField(const BSONObj& b, const std::string& field)
+{
+  if (!b.hasField(field))
+  {
+    LM_E(("Runtime Error (object field '%s' is missing in BSONObj <%s>)", field.c_str(), b.toString().c_str()));
+    return BSONObj();
+  }
+  if (b.getField(field).type() != Object)
+  {
+
+    LM_E(("Runtime Error (field '%s' was supposed to be an object but type=%d in BSONObj <%s>)",
+          field.c_str(), b.getField(field).type(), b.toString().c_str()));
+    return BSONObj();
+  }
```

Comment:
Make sense... I first do some test with that approach and, if works, I will use it.